### ISSUE TITLE
The fused-CE wrapper drops family logit transforms on its non-fused branch

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#1048]
## Problem

`c0f95ced` (#988) changed the gate at `_factory.py:374-376` from `kwargs.get("fused_linear_cross_entropy", False)` to inherit from `performance`, and `_dp_trainer.py:844-850` always calls `apply_model_patches(..., performance=True)` — so the fused causal-LM forward wrapper is now installed on **all seventeen families** in `_FUSED_CE_CAUSAL_LM` (`cross_entropy.py:122-140`) **by default**.

The wrapper's fused branch resolves and threads the family transform into the kernels (`cross_entropy.py:308-323, 336`). But when `use_fused_ce` (`:282-292`) is False, the fallback branch at `:356-364` does **not** delegate to `original(...)` — it recomputes logits itself:

```python
logits = self.lm_head(hidden_states[:, slice_indices, :])
loss = self.loss_function(logits, labels, self.vocab_size, **kwargs)
```

That skips the family's post-`lm_head` transform entirely: Cohere/Cohere2 `logit_scale`, Granite `logits_scaling`, Gemma2 `final_logit_softcapping`.

## Impact

Four families train and evaluate on differently-scaled logits than upstream, on a default-on code path, with no error. For Gemma2 the missing softcap also changes the loss landscape rather than just its scale.

The wrapper already does the right thing two branches away — it delegates to `original(...)` at `:204-220` and for MoE router logits at `:225-244` — so this is an omission, not a design constraint.

## Acceptance criteria

- The fallback branch delegates to `original(self, ...)`; it already has every argument it needs. (Nearly free, and the preferred fix.)
- Failing that, hoist the `logit_scale` / `logits_scaling` / `final_logit_softcapping` resolution above `if use_fused_ce:` and apply it to `logits` before calling `self.loss_function`.
- `test_fused_ce_text_family_parity.py` extends to Cohere, Cohere2, Granite and Gemma2 over **both** branches — its fixtures are currently olmo2/olmo3/smollm3/ministral, none of which has a post-`lm_head` transform, which is why this shipped.
- The related fused DFT evaluation path is covered too: it needs the scale threaded into `linear_nll_sum` / `fused_dft_loss`, so gate `_fused_dft` off for scaled families in the same PR and thread the scale behind the routing decision.

Audit source: OPQ-460, OPQ-461 (merged).
